File: src/core/command_handlers/favorites.py
class FavoriteCommands:
    @staticmethod
    def handle_fav(chat_manager, ui, parts):
        favorites = chat_manager.config_manager.get("favorites", [])
        if len(parts) == 1:
            if not favorites:
                ui.display_info(
                    "No favorite models saved. Use /fav+ to add the current one."
                )
                return True
            header = "| ID | Model (Provider) | Temp |\n|---|---|---|\n"
            rows = []
            for i, f in enumerate(favorites):
                m = f"{f['model']} ({f['provider']})"
                t = f["temp"]
                rows.append(f"| {i} | {m} | {t} |")
            ui.display_panel("Favorite Models", header + "\n".join(rows), style="green")
            return True
        elif parts[1] == "+":
            provider = chat_manager.config_manager.get("provider", "Gemini")
            model = chat_manager.assistant._current_model
            temp = chat_manager.temperature
            new_fav = {"provider": provider, "model": model, "temp": temp}
            if new_fav not in favorites:
                favorites.append(new_fav)
                chat_manager.save_config("favorites", favorites)
                ui.display_info(f"Added to favorites: {provider} | {model}")
            else:
                ui.display_info("Model already in favorites.")
            return True
        elif parts[1].startswith("-"):
            try:
                idx = int(parts[1][1:])
                if 0 <= idx < len(favorites):
                    removed = favorites.pop(idx)
                    chat_manager.save_config("favorites", favorites)
                    ui.display_info(f"Removed from favorites: {removed['model']}")
                else:
                    ui.display_error("Invalid index.")
            except (ValueError, IndexError):
                ui.display_error("Usage: /fav -<index>")
            return True
        else:
            try:
                idx = int(parts[1])
                if 0 <= idx < len(favorites):
                    fav = favorites[idx]
                    from src.infrastructure.gemini.adapter import GeminiAdapter
                    from src.infrastructure.nvidia.adapter import NvidiaAdapter

                    providers = {"Gemini": GeminiAdapter, "NVIDIA": NvidiaAdapter}
                    provider_name = fav["provider"]
                    if provider_name not in providers:
                        ui.display_error(f"Unknown provider: {provider_name}")
                        return True
                    if chat_manager.config_manager.get("provider") != provider_name:
                        chat_manager.assistant = providers[provider_name]()
                        chat_manager.save_config("provider", provider_name)
                    chat_manager.save_config("model_name", fav["model"])
                    chat_manager.save_config("default_temperature", fav["temp"])
                    chat_manager.temperature = fav["temp"]
                    chat_manager.assistant.start_chat(
                        fav["model"], chat_manager.system_instruction, fav["temp"]
                    )
                    ui.display_info(
                        f"Loaded favorite {idx}: {provider_name} | {fav['model']} | Temp: {fav['temp']}"
                    )
                else:
                    ui.display_error("Favorite index not found.")
            except (ValueError, IndexError):
                ui.display_error("Usage: /fav <index> or /fav+ or /fav -<index>")
            return True

File: src_js/core/commandHandlers/favorites.js
import { GeminiAdapter } from '../../infrastructure/gemini/adapter.js';

export class FavoriteCommands {
  static async handleFav(chatManager, ui, parts) {
    let favorites = chatManager.configManager.get('favorites', []);

    if (parts.length === 1) {
      if (favorites.length === 0) {
        ui.displayInfo('No favorite models saved. Use /fav+ to add the current one.');
        return true;
      }

      let header = '| ID | Model (Provider) | Temp |\n|---|---|---|\n';
      let rows = [];
      for (let i = 0; i < favorites.length; i++) {
        const f = favorites[i];
        const m = `${f.model} (${f.provider})`;
        const t = f.temp;
        rows.push(`| ${i} | ${m} | ${t} |`);
      }
      ui.displayMessage('--- Favorite Models ---\n' + header + rows.join('\n'));
      return true;
    }

    if (parts[1] === '+') {
      const provider = chatManager.configManager.get('provider', 'Gemini');
      const model = chatManager.assistant.currentModel;
      const temp = chatManager.temperature;
      const newFav = { provider, model, temp };

      if (!favorites.some(f => f.model === model && f.provider === provider)) {
        favorites.push(newFav);
        chatManager.configManager.save('favorites', favorites);
        ui.displayInfo(`Added to favorites: ${provider} | ${model}`);
      } else {
        ui.displayInfo('Model already in favorites.');
      }
      return true;
    }

    if (parts[1].startsWith('-')) {
      try {
        const idx = parseInt(parts[1].slice(1), 10);
        if (idx >= 0 && idx < favorites.length) {
          const removed = favorites.splice(idx, 1)[0];
          chatManager.configManager.save('favorites', favorites);
          ui.displayInfo(`Removed from favorites: ${removed.model}`);
        } else {
          ui.displayError('Invalid index.');
        }
      } catch (e) {
        ui.displayError('Usage: /fav -<index>');
      }
      return true;
    }

    try {
      const idx = parseInt(parts[1], 10);
      if (idx >= 0 && idx < favorites.length) {
        const fav = favorites[idx];
        const providerName = fav.provider;
        const currentProvider = chatManager.configManager.get('provider', 'Gemini');

        if (providerName !== currentProvider) {
          if (providerName === 'Gemini') {
            const { GeminiAdapter } = await import('../../infrastructure/gemini/adapter.js');
            chatManager.assistant = new GeminiAdapter();
          } else if (providerName === 'NVIDIA') {
            const { NvidiaAdapter } = await import('../../infrastructure/nvidia/adapter.js');
            chatManager.assistant = new NvidiaAdapter();
          } else {
            ui.displayError(`Provider ${providerName} not supported.`);
            return true;
          }
          chatManager.configManager.save('provider', providerName);
        }

        chatManager.configManager.save('model_name', fav.model);
        chatManager.configManager.save('default_temperature', fav.temp);
        chatManager.temperature = fav.temp;
        chatManager.assistant.startChat(fav.model, chatManager.systemInstruction, fav.temp);

        ui.displayInfo(`Loaded favorite ${idx}: ${providerName} | ${fav.model} | Temp: ${fav.temp}`);
      } else {
        ui.displayError('Favorite index not found.');
      }
    } catch (e) {
      ui.displayError('Usage: /fav <index> or /fav+ or /fav -<index>');
    }
    return true;
  }
}
